src/ClientData/include/ClientData/CallstackData.h

```diff
@@ -83,9 +83,32 @@ class CallstackData {
   template <typename Action>
   void ForEachCallstackEventInTimeRangeDiscretized(uint64_t min_timestamp, uint64_t max_timestamp,
                                                    uint32_t resolution, Action&& action) const {
-    for (const auto& [tid, unused_events] : callstack_events_by_tid_) {
-      ForEachCallstackEventOfTidInTimeRangeDiscretized(tid, min_timestamp, max_timestamp,
-                                                       resolution, action);
+    auto get_next_callstack = [&](uint64_t timestamp) -> CallstackEvent {
+      constexpr uint32_t kFakeId = -1;
+      CallstackEvent next_callstack{max_timestamp, kFakeId, kFakeId};
+      uint32_t current_pixel = GetPixelNumber(timestamp, resolution, min_timestamp, max_timestamp);
+      for (const auto& [unused_tid, events] : callstack_events_by_tid_) {
+        auto next_callstack_of_tid = events.lower_bound(timestamp);
+        if (next_callstack_of_tid != events.end() &&
```

Comment:
Propose to simplify the flow by splitting the nested ifs into several conditions like:
- if the callstack candidate is end, continue
- if the callstack candidate is within the pixel, return it
- if the callstack candidate is earlier than the next callstack, set the next callstack to it

This would make the flow a bit easier to follow